src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs

```diff
@@ -146,7 +146,7 @@ public void Equals_Object_ReturnsExpected(GCHandle handle, object other, bool ex
             finally
             {
                 handle.Free();
-                if (other is GCHandle otherHandle)
+                if (other is GCHandle otherHandle && !expected)
```

Comment:
When you have two GCHandles and you expect them to be the same, you only free one of them. Technically you can change the check to be using Equals, but in practice it makes no difference.

BTW, there shouldn't be a case valid GCHandle is equal to another valid GCHandle. GCHandle is basically an pointer to somewhere in the runtime's internal table. 
It is a value type, so using ReferenceEquals is not right either (it does boxing).